test-e2e/specs/anonymous-recordSpec.js

```diff
@@ -53,21 +53,21 @@ describe( 'anonymous record', function() {
     });
     
     it( 'sets recordA', function(done) {
-        anonymousRecord.setName( 'recordA' );
-        setTimeout(function(){
-            expect( currentPet ).toBe( 'hamster' );
+        anonymousRecord.once( 'ready', function(){
             expect( anonymousRecord.get() ).toEqual({ pet: 'hamster' });
+            expect( currentPet ).toBe( 'hamster' );
             done();
-        }, 20);
+        });
+        anonymousRecord.setName( 'recordA' );
     });
     
     it( 'sets recordB', function(done) {
-        anonymousRecord.setName( 'recordB' );
-        setTimeout(function(){
-            expect( currentPet ).toBe( 'pug' );
+        anonymousRecord.once( 'ready', function(){
```

Comment:
Would be good to add a test that makes sure that this happens exactly once. I'll add one on a unit test level